data/restapi/src/main/java/tech/pegasys/artemis/restapi/RestApi.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright 2019 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.artemis.restapi;
+
+import static tech.pegasys.artemis.provider.JsonProvider.objectToJSON;
+
+import com.google.common.primitives.UnsignedLong;
+import io.javalin.Javalin;
+import java.util.HashMap;
+import java.util.Map;
+import org.apache.tuweni.bytes.Bytes32;
+import tech.pegasys.artemis.datastructures.blocks.BeaconBlock;
+import tech.pegasys.artemis.datastructures.state.BeaconState;
+import tech.pegasys.artemis.networking.p2p.JvmLibP2PNetwork;
+import tech.pegasys.artemis.networking.p2p.api.P2PNetwork;
+import tech.pegasys.artemis.storage.ChainStorageClient;
+
+public class RestApi {
+
+  private Javalin app;
+  private ChainStorageClient chainStorageClient;
+  private JvmLibP2PNetwork network;
+  boolean isLibP2P = false;
+
+  public RestApi(
+      ChainStorageClient chainStorageClient, P2PNetwork p2pNetwork, final int portNumber) {
+    this.chainStorageClient = chainStorageClient;
+    if (p2pNetwork instanceof JvmLibP2PNetwork) {
+      isLibP2P = true;
+      network = (JvmLibP2PNetwork) p2pNetwork;
+    }
+    app = Javalin.create().start(portNumber);
+
+    setGenesisTimeHandler();
+    setBeaconHeadHandler();
+    setBeaconBlockHandler();
+    setBeaconStateHandler();
+    setPeerIDHandler();
+    setPeersHandler();
+    setENRHandler();
```

Comment:
Given that the number of APIs will grow over time, it would probably be better to make each its own class, probably with a common interface that is then registered here.  That would also let us test each method individually.